File: spec/targets/polyglot/skeletons/c/c23/stunir_runtime.c
#include "stunir_runtime.h"

#include <stdio.h>

void stunir_println(const char* s) {
  if (s == NULL) {
    fputs("\n", stdout);
    return;
  }
  fputs(s, stdout);
  fputs("\n", stdout);
}

int stunir_program(void) {
  /* STUB: codegen should replace this body. */
  stunir_println("STUNIR C skeleton: program stub");
  return 0;
}

File: spec/targets/polyglot/skeletons/erlang/rebar3/src/stunir_runtime.erl
-module(stunir_runtime).
-export([println/1, program/0]).

println(S) when is_list(S) ->
    io:format("~s~n", [S]);
println(S) ->
    io:format("~p~n", [S]).

program() ->
    %% STUB: codegen should replace this body.
    println("STUNIR Erlang skeleton: program stub"),
    0.
